src/mem/ruby/slicc_interface/AbstractEntry.hh

```diff
@@ -31,14 +31,15 @@
 
 #include <iostream>
 
-#include "mem/protocol/AccessPermission.hh"
+//#include "mem/protocol/AccessPermission.hh"
+#include "mem/ruby/slicc_interface/AbstractCacheEntry.hh"
 
-class AbstractEntry
+class AbstractEntry : public AbstractCacheEntry
 {
   public:
     AbstractEntry();
     virtual ~AbstractEntry() = 0;
-
+/*
```

Comment:
We should just remove the commented out part.

Also, don't we still want print() -- I guess inherit it, but I assume Directory would want it's own version?